### PR TITLE
feat(shared-docs): create stackblitz post API pages generator

### PR DESCRIPTION
**Pull request by josephperrott**
A script and associated bazel rule for creating a stackblitz post API html page for examples.

### CODE REVIEW

File: docs/markdown/_stackblitz.bzl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    return [DefaultInfo(files = depset([html_output]))]
+
+generate_stackblitz = rule(
+    # Point to the starlark function that will execute for this rule.
+    implementation = _generate_stackblitz,
+    doc = """Rule that renders markdown sources to html""",
+
+    # The attributes that can be set to this rule.
+    attrs = {
+        "example": attr.label(
```
Comment (jelbourn) [inline docs/markdown/_stackblitz.bzl:55]: Maybe `example_srcs` ?

**Review by jelbourn — APPROVED** [review]
LGTM

File: docs/markdown/examples/stackblitz/file-system.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {existsSync} from 'fs';
+import {copyFile, mkdir, readdir, rm, stat} from 'fs/promises';
+import {join} from 'path';
+
+/**
+ * Recurusivelt copy folder and contents to the destigation, creating the destination folder
```
Comment (devversion) [inline docs/markdown/examples/stackblitz/file-system.ts:14]: Nit: typo

File: docs/markdown/examples/stackblitz/file-system.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * if necessary.
+ **/
```
Comment (devversion) [inline docs/markdown/examples/stackblitz/file-system.ts:16]: Why can't we use FS copy methods? they seem to support recursive copies?

File: docs/markdown/examples/stackblitz/builder.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      form.appendChild(element);
+    }
+  }
+
+  return doc.documentElement.outerHTML;
+}
+
+function createBaseStackblitzHtml(primaryFile: string) {
+  const file = `?file=${primaryFile}`;
+  const action = `https://stackblitz.com/run${file}`;
```
Comment (devversion) [inline docs/markdown/examples/stackblitz/builder.ts:145]: could that be just inlined? (a separate variable seems unnecessary)

File: docs/markdown/_stackblitz.bzl [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            doc = """Files used for the stackblitz generation.""",
+        ),
+        "stackblitz_template": attr.label(
```
Comment (devversion) [inline docs/markdown/_stackblitz.bzl:58]:
```suggestion
        "template_srcs": attr.label(
```

**Review by devversion — APPROVED** [review]

File: docs/markdown/examples/stackblitz/file-system.ts (hunk 3 above)
Comment (devversion) [reply docs/markdown/examples/stackblitz/file-system.ts:18]: sounds good. created an issue so that we don't forget about this

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 1337ad7c046566310c1748a4bdf3a12d89207a0d.

File: docs/markdown/_stackblitz.bzl [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
-    """Implementation of the markdown rule"""
+    """Implementation of the stackblitz generator rule"""
 
     # Determine the stackblitz template base directory
-    stackblitz_template = " " * 1000
-    for file in ctx.files.stackblitz_template:
+    # Create an initial string of length greater than the length the stackblitz template path will
+    # be. This is used so that the first file path we encounter is used as the initial value.
+    template_srcs = " " * 10000
+    for file in ctx.files.template_srcs:
```
Comment (devversion) [inline docs/markdown/_stackblitz.bzl:10]: Have we considered specifying a base path, and simply having a default here? It should be "short paths" anyway. I'm thinking in some cases the template _could_ intentionally be nested deeper in the "final result".

File: docs/markdown/_stackblitz.bzl [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
 def _generate_stackblitz(ctx):
-    """Implementation of the markdown rule"""
+    """Implementation of the stackblitz generator rule"""
 
     # Determine the stackblitz template base directory
-    stackblitz_template = " " * 1000
-    for file in ctx.files.stackblitz_template:
+    # Create an initial string of length greater than the length the stackblitz template path will
+    # be. This is used so that the first file path we encounter is used as the initial value.
+    template_srcs = " " * 10000
```
Comment (devversion) [inline docs/markdown/_stackblitz.bzl:9]: Nit: A more robust approach would be to use `None` and check it below as an `or` case. And then have a safety check at the end to make sure we actually saw some sources

**Review by devversion — APPROVED** [review]
LGTM. Two minor nits

**Pull request closed without merging**